openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -649,114 +605,116 @@ def _get_library_component_tags_count(library_key) -> dict:
-    library_component_tags_count = _get_library_component_tags_count(library_key)
-
-    for usage_key in usages:
-        # For top-level definitions, we can go from definition key to usage key using the following, but this would
-        # not work for non-top-level blocks as they may have multiple usages. Top level blocks are guaranteed to
-        # have only a single usage in the library, which is part of the definition of top level block.
-        def_key = lib_bundle.definition_for_usage(usage_key)
-        display_name = get_block_display_name(def_key)
-        text_match = (text_search is None or
-                      text_search.lower() in display_name.lower() or
-                      text_search.lower() in str(usage_key).lower())
-        type_match = (block_types is None or usage_key.block_type in block_types)
-        if text_match and type_match:
-            metadata.append({
-                "id": usage_key,
-                "def_key": def_key,
-                "display_name": display_name,
-                "has_unpublished_changes": lib_bundle.does_definition_have_unpublished_changes(def_key),
-                "tags_count": library_component_tags_count.get(str(usage_key), 0),
-            })
+    Get the library blocks and filter.
 
+    TODO: This is primarily used in tests, but it's getting an unbounded list,
+    instead of a pagable queryset. We should get rid of this function
+    altogether.
+    """
     return [
-        LibraryXBlockMetadata(
-            usage_key=item['id'],
-            def_key=item['def_key'],
-            display_name=item['display_name'],
-            has_unpublished_changes=item['has_unpublished_changes'],
-            tags_count=item['tags_count']
-        )
-        for item in metadata
+        LibraryXBlockMetadata.from_component(library_key, component)
+        for component in get_library_components(library_key, text_search, block_types)
     ]
 
 
-def _lookup_usage_key(usage_key) -> tuple[BundleDefinitionLocator, LibraryBundle]:
+def get_library_components(library_key, text_search=None, block_types=None) -> QuerySet[Component]:
     """
-    Given a LibraryUsageLocatorV2 (usage key for an XBlock in a content library)
-    return the definition key and LibraryBundle
-    or raise ContentLibraryBlockNotFound
+    Get the library components and filter.
+
+    TODO: Full text search needs to be implemented as a custom lookup for MySQL,
+    but it should have a fallback to still work in SQLite.
     """
-    assert isinstance(usage_key, LibraryUsageLocatorV2)
-    lib_context = get_learning_context_impl(usage_key)
-    def_key = lib_context.definition_for_usage(usage_key, force_draft=DRAFT_NAME)
-    if def_key is None:
-        raise ContentLibraryBlockNotFound(usage_key)
-    lib_bundle = LibraryBundle(usage_key.lib_key, def_key.bundle_uuid, draft_name=DRAFT_NAME)
-    return def_key, lib_bundle
+    lib = ContentLibrary.objects.get_by_key(library_key)  # type: ignore[attr-defined]
```

Comment:
@kdmccormick: This is me giving up. I belatedly realized that the original call before my conversion also had this "type: ignore" on it.

Reply:
Will be handled in https://github.com/openedx/edx-platform/pull/34237